contracts/token/ERC721/extensions/ERC721URIStorage.sol

```diff
@@ -48,15 +48,16 @@ abstract contract ERC721URIStorage is ERC721 {
     }
 
     /**
-     * @dev See {ERC721-_burn}. This override additionally checks to see if a
+     * @dev See {ERC721-_update}. When burning this override additionally checks to see if a
      * token-specific URI was set for the token, and if so, it deletes the token URI from
      * the storage mapping.
      */
-    function _burn(uint256 tokenId) internal virtual override {
-        super._burn(tokenId);
-
-        if (bytes(_tokenURIs[tokenId]).length != 0) {
-            delete _tokenURIs[tokenId];
+    function _update(address from, address to, uint256 firstTokenId, uint256 batchSize) internal virtual override {
+        super._update(from, to, firstTokenId, batchSize);
+        if (to == address(0)) {
```

Comment:
if `to == address(0)` and `batchSize > 1`, we should probably do something. Revert?